### ISSUE TITLE
Blockchain Synced, Still eth.blockNumber returns 0

### ISSUE THREAD

**Issue by prashantprabhakar** [ethereum/go-ethereum#18507]
I am using ubuntu 16.04 to sync geth node in fast mode. I know geth node is difficult to keep up with HDD so I am using following server configuration.

RAM: 4GB
Disk: 500GB Provisional SSD with 100 IOPS
CPUs: 2 

#### System information

Geth version: `1.8.21-stable`
OS & Version: Linux ( Ubuntu 16.04)

#### Expected behaviour
Balances should reflect and block number should be latest one.

#### Actual behaviour
eth.blockNumber returns 0 and hence no balances are refleceted.

#### Steps to reproduce the behaviour
eth.syncing results
```
{
  currentBlock: 7113789,
  highestBlock: 7113858,
  knownStates: 128401737,
  pulledStates: 128398634,
  startingBlock: 7113353
}

```

Current block is only 69 blocks behind the highest blocks. It has been 4 days since I started syncing the blockchain. 

Geth syncing is really a pain. We are not sure how much time it will take to sync. If I need to restart syncing  by downgrading geth or wait for this sync. There is no definite time we can give to users when the BC will be available. How do we know if we are synced or still resolving state entries.

#### Geth Logs

seems geth has imported state entries, since long trail of importing state entries is gone.
```
NFO [01-23|13:08:32.010] Imported new block headers               count=1    elapsed=4.964ms   number=7113921 hash=ffeae4…9fdce2
INFO [01-23|13:08:38.556] Imported new block headers               count=1    elapsed=4.744ms   number=7113922 hash=589762…9303f1
INFO [01-23|13:08:51.380] Imported new block headers               count=1    elapsed=5.203ms   number=7113923 hash=92141e…f7d106
INFO [01-23|13:08:53.766] Imported new state entries               count=1148 elapsed=4.160ms   processed=1

```

**Comment by prashantprabhakar** [comment]
Yeah, I know states have not been downloaded. But why not is the problem. Network speed is 0.25 - 0.65 GiB/sec. CPU utilization is hardly to 10%. I do keep checking logs, do not see lots of peer dropping. 
Another day  has passed still not synced yet.

**Comment by byzh4** [comment]
why not can get blockNumber  then syncing ?
